### PR TITLE
Filesystem: add support for F2FS

### LINKED ISSUE #40418: Add support for f2fs to filesystems
##### ISSUE TYPE
 - Feature Idea

##### COMPONENT NAME

filesystem

##### ANSIBLE VERSION

```
ansible 2.5.3
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/home/langhorst/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/dist-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.12 (default, Dec  4 2017, 14:50:18) [GCC 5.4.0 20160609]
```

##### CONFIGURATION

no output

##### OS / ENVIRONMENT

linux

##### SUMMARY

cannot make an f2fs filesystem from ansible
```
failed: [seq-flash] (item={u'name': u'md0', u'level': 0, u'devices': [u'/dev/nvme0n1p5', u'/dev/nvme1n1p1'], u'state': u'present', u'filesystem': u'f2fs', u'mountpoint': u'/seq_data'}) => {"changed": false, "item": {"devices": ["/dev/nvme0n1p5", "/dev/nvme1n1p1"], "filesystem": "f2fs", "level": 0, "mountpoint": "/seq_data", "name": "md0", "state": "present"}, "msg": "value of fstype must be one of: ext4, ext3, ext2, btrfs, vfat, ext4dev, reiserfs, lvm, xfs, got: f2fs"}
```
##### STEPS TO REPRODUCE

```yaml
  - include_role:
      name: mrlesmithjr.mdadm
    vars:
      mdadm_arrays:
       - name: md0
         devices: [/dev/nvme0n1p5, /dev/nvme1n1p1]
         filesystem: f2fs
         level: 0
         mountpoint: /seq_data
         state: present
```

### PR DESCRIPTION
**Pull request by pilou-**
##### SUMMARY
Filesystem: add support for F2FS

Integration tests provided (Ubuntu 16.04, Fedora (all versions), OpenSuse)

Fixes #40418

##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
filesystem

##### ANSIBLE VERSION
```
ansible 2.6.0 (devel 44d9dd2c77) last updated 2018/05/20 14:41:59 (GMT +200)
```

##### ADDITIONAL INFORMATION

grow/resize operation requires quite a recent version of f2fs-tools.

### CODE REVIEW

**Comment by pilou-** [comment]
indentation doesn't contain tabs: a comment contains a tab on purpose

**Comment by mattclay** [comment]
@pilou- I've opened https://github.com/ansible/ansible/pull/40509 to remove the tab check from the module validator, since pycodestyle performs the same check without the false positive.

I've also removed the broken label from this PR. This isn't a bot issue -- it's just commenting with the results of the Shippable CI run.

**Comment by pilou-** [comment]
@mattclay thanks !

**Comment by mattclay** [comment]
CI failure in integration tests: https://app.shippable.com/github/ansible/ansible/runs/66523/57/tests

**Comment by pilou-** [comment]
Pull request updated:
1. rebased
2. s/f2fs support was added in 2.6/f2fs support was added in 2.7/

File: lib/ansible/modules/system/filesystem.py [hunk 1]
```diff
@@ -233,6 +234,49 @@ class Ocfs2(Filesystem):
...
+        return ''
+
+    def get_fs_size(self, dev):
+        cmd = self.module.get_bin_path('dump.f2fs', required=True)
+        # Get sector count and sector size
+        _, dump, _ = self.module.run_command([cmd, str(dev)], check_rc=True)
+        sector_size = None
+        sector_count = None
+        for line in dump.splitlines():
+            if 'Info: sector size = ' in line:
```
Comment (mscherer) [inline lib/ansible/modules/system/filesystem.py]: The locale should be set to C if we do string matching.

Comment (pilou-) [reply lib/ansible/modules/system/filesystem.py]: Done: `LANG=C` has been added here and for other cases where command output is parsed.

**Comment by mscherer** [comment]
shipit

**Pull request merged**